src/test/ProposalPayloadE2E.t.sol

```diff
@@ -3,37 +3,335 @@ pragma solidity ^0.8.15;
+        assertEq(liqRate, 0);
+        assertEq(stableRate, 3 * (AaveV2Helpers.RAY / 100));
+        assertEq(varRate, 3 * (AaveV2Helpers.RAY / 100));
+    }
+
+    function testUtilizationAtOneHundredPercent() public {
+        (uint256 liqRate, uint256 stableRate, uint256 varRate) = strategy.calculateInterestRates(
+            BAL,
+            0,
+            0,
+            100 * 1e18,
+            68200000000000000000000000,
+            2000
+        );
+
+        // At max borrow rate, stable rate should be 87% and variable rate should be 85.75%.
+        assertEq(liqRate, 1336000000000000000000000000);
+        assertEq(stableRate, 3130000000000000000000000000);
+        assertEq(varRate, 1670000000000000000000000000);
+    }
+
+    function testUtilizationAtUOptimal() public {
+        (uint256 liqRate, uint256 stableRate, uint256 varRate) = strategy.calculateInterestRates(
+            BAL,
+            20 * 1e18,
+            0,
+            80 * 1e18,
+            68200000000000000000000000,
+            2000
+        );
+
+        // At UOptimal, stable rate should be % and variable rate should be 14%.
+        assertEq(liqRate, 108800000000000000000000000);
+        assertEq(stableRate, 13 * (AaveV2Helpers.RAY / 100));
+        assertEq(varRate, 17 * (AaveV2Helpers.RAY / 100));
+    }
+
+    function _getExpectedStrategyPolygon() public view returns (InterestStrategyValues memory) {
+        return
+            InterestStrategyValues({
+                addressesProvider: address(OLD_INTEREST_RATE_STRATEGY_V3_POLYGON.ADDRESSES_PROVIDER()),
+                optimalStableToTotalDebtRatio: 20 * (RAY / 100),
+                optimalUsageRatio: 80 * (RAY / 100),
+                baseStableBorrowRate: 0,
+                baseVariableBorrowRate: 0,
+                stableRateSlope1: OLD_INTEREST_RATE_STRATEGY_V3_POLYGON.getStableRateSlope1(),
+                stableRateSlope2: 80 * (RAY / 100),
+                variableRateSlope1: 575 * (RAY / 10000),
+                variableRateSlope2: 80 * (RAY / 100)
+            });
+    }
+
+    function _modifyStrategy() internal {
         // Pass vote and execute proposal
+        vm.selectFork(mainnetFork);
+        vm.recordLogs();
         GovHelpers.passVoteAndExecute(vm, proposalId);
 
-        // Post-execution assertations
+        Vm.Log[] memory entries = vm.getRecordedLogs();
+        assertEq(keccak256("StateSynced(uint256,address,bytes)"), entries[5].topics[0]);
+        assertEq(address(uint160(uint256(entries[5].topics[2]))), FX_CHILD_ADDRESS);
+
+        // Mock the receive on L2 with the data emitted on StateSynced
+        vm.selectFork(polygonFork);
+        vm.startPrank(BRIDGE_ADMIN);
+        IStateReceiver(FX_CHILD_ADDRESS).onStateReceive(uint256(entries[5].topics[1]), this._cutBytes(entries[5].data));
+        vm.stopPrank();
+
+        // Forward time & execute proposal
+        BridgeExecutorHelpers.waitAndExecuteLatest(vm, POLYGON_BRIDGE_EXECUTOR);
     }
```

Comment:
Would recommend moving both these utility functions to right after the `setUp()` function so that it's easier to find.